crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -156,6 +156,11 @@ impl CheckpointExecutor {
                 .check_epoch_last_checkpoint(epoch_store.clone(), &highest_executed)
                 .await
             {
+                self.checkpoint_store
+                    .prune_local_summaries()
```

Comment:
No strong opinions, but couldn't we also just move this table to epoch store?

Reply:
+1
`locally_computed_checkpoints` is not useful across epoch boundary with the new fork detection logic.

Reply:
we don't have access to the epoch store when state sync writes a certified checkpoint.

(note that i did add pruning for this table)

Reply:
That tells me that we should move the fork detect call to within checkpoint aggregator, when we insert certified checkpoint.

Reply:
but that is not the only place that certified checkpoints are inserted. they can be inserted via state sync, which is why i put this check where i did